lib/turkish_cities/district.rb

```diff
@@ -26,21 +26,28 @@ def list_neighborhoods(city_name, district_name, subdistrict_name)
     district_list = create_district_list(city_name)
 
     if !district_list[district_name].nil?
-      neighborhoods = if subdistrict_name.nil?
-                        create_neighborhoods_without_subdistrict_name(district_list, district_name)
-                      else
-                        create_neighborhoods_with_subdistrict_name(district_list,
-                                                                   district_name,
-                                                                   subdistrict_name)
-                      end
-      sort_alphabetically(neighborhoods)
+      neighborhoods = create_neighborhoods(district_list, city_name,
+                                           district_name, subdistrict_name)
+      neighborhoods.is_a?(Array) ? sort_alphabetically(neighborhoods) : neighborhoods
     else
       district_not_found_error(district_name, city_name)
     end
   end
 
   private
 
+  def create_neighborhoods(district_list, city_name, district_name, subdistrict_name)
```

Comment:
bu method kafami yakti :) nasil duzeltebiliriz? cok fazla parametre almis. belki oradan baslamak lazim.

Reply:
@enderahmetyurt Kusura bakma bir kaç haftadır bakamadım repo'ya anca vakit bulabildim. Benim de aklıma bir şey gelmedi kolaylaştırmak için. Unutulmasın diye issue açtım #45